packages/theme-language-server-common/src/completions/providers/FilterNamedParameterCompletionProvider.ts

```diff
@@ -0,0 +1,57 @@
+import { NodeTypes } from '@shopify/liquid-html-parser';
+import { CompletionItem, CompletionItemKind } from 'vscode-languageserver';
+import { CURSOR, LiquidCompletionParams } from '../params';
+import { Provider, createCompletionItem } from './common';
+import { ThemeDocset } from '@shopify/theme-check-common';
+
+export class FilterNamedParameterCompletionProvider implements Provider {
+  constructor(private readonly themeDocset: ThemeDocset) {}
+
+  async completions(params: LiquidCompletionParams): Promise<CompletionItem[]> {
+    if (!params.completionContext) return [];
+
+    const { node } = params.completionContext;
+    if (!node || node.type !== NodeTypes.VariableLookup) {
+      return [];
+    }
+
+    if (!node.name || node.lookups.length > 0) {
+      // We only do top level in this one.
+      return [];
+    }
+
+    const partial = node.name.replace(CURSOR, '');
+    const currentContext = params.completionContext.ancestors.at(-1);
+
+    if (!currentContext || currentContext?.type !== NodeTypes.LiquidFilter) {
+      return [];
+    }
+
+    const filters = await this.themeDocset.filters();
+    const foundFilter = filters.find((f) => f.name === currentContext.name);
+
+    if (!foundFilter?.parameters) {
+      return [];
+    }
+
+    const options = foundFilter.parameters
+      .filter((p) => p.name.startsWith(partial))
+      .map((parameter) => {
+        return {
+          entry: {
+            ...parameter,
+          },
+          types: parameter.types,
+        };
+      });
+
+    return options.map(({ entry, types }) =>
+      createCompletionItem(
+        entry,
```

Comment:
Is there any limitations today that prevent us from putting a ":" after the param? It would be great if we can do something like `${param}: $0`.

Reply:
There is! Not all params are named and will require a `:`. I'm working on updating the source of the docs to include this so that we can do that!